### PR TITLE
Adding Export Compliance tab and fetching quay user details (PROJQUAY-4173)

### PR DESCRIPTION
**Pull request by Sunandadadi**
This PR:
1. Adds role checks and displays various components accordingly. There are `ADMIN_ROLE` and `EXPORT_COMPLIANCE_ROLE` defined in the config.yaml
2. Adds a new tab `Export Compliance` that fetches quay user data from the supplied Red Hat SSO username.
3. Adds a new permission check decorator for all backend endpoints.

Users with only export compliance role see the export compliance tab. The export compliance tab has an option to fetch  quay user data from the supplied Red Hat SSO username and clean build queue & disable the user.
Screen Shot 2022-10-05 at 12 44 36 PM
Screen Shot 2022-10-05 at 12 44 52 PM

Users with admin and export compliance role see all the tabs of the tool:
Screen Shot 2022-10-05 at 12 48 50 PM

### CODE REVIEW

File: frontend/src/app/UserUtils/actions/FetchUserFromEmail.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+export const FetchUserFromEmail: React.FunctionComponent = (props) => {
+  const [userEmail, setUserEmail] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [message, setMessage] = useState('');
+  const [response, setResponse] = useState(false);
+
+  const userEmailOnChange = (value) => {
+    setUserEmail(value);
+    if (value == '') {
+      setResponse('');
```
[the remark this answers is not in the document]
Comment (bcaton85) [reply frontend/src/app/UserUtils/actions/FetchUserFromEmail.tsx:21]: also not sure if i would expect the content to be removed when i clear the field

**Review by bcaton85 — COMMENTED** [review]
Great work, let some small comments - nothing big.

File: backend/utils.py [hunk 2]
```diff
@@ -67,3 +70,46 @@ def wrapper(*args, **kwargs):
...
+
+
+def verify_admin_or_export_perm(func):
+    def wrapper(*args, **kwargs):
+        if not current_user or not current_user.realm_access:
+            return make_response(json.dumps({"message": "No RBAC defined for user"}), 401)
+
+        ADMIN_ROLE = app.config.get('authentication', {}).get('roles', {}).get('ADMIN_ROLE')
+        EXPORT_COMPLIANCE_ROLE = app.config.get('authentication', {}).get('roles', {}).get('EXPORT_COMPLIANCE_ROLE')
+        if not (set([ADMIN_ROLE, EXPORT_COMPLIANCE_ROLE]) & set(current_user.realm_access.get('roles', None))):
```
[the remark this answers is not in the document]
Comment (Sunandadadi) [reply backend/utils.py:110]: So, that was my first thought too. But, I'd need to hardcode the permissions for that. I cannot read it from the app.config in a decorator. Something like:

```
@verify_permissions(perms=[app.config.get('authentication', {}).get('roles', {}).get('ADMIN_ROLE')])
def func()
```

returns out of context flask error. So, either I create separate decorators or hard-code the permissions. But, didn't like the idea of exposing permissions so created separate decorators.

File: frontend/src/app/SiteUtils/SiteUtils.tsx [hunk 3]
```diff
@@ -181,6 +186,7 @@ export const SiteUtils: React.FunctionComponent = (props) => {
   }
 
   return (
+    ( UserService.hasRealmRole(ADMIN_ROLE) ?
```
[the remark this answers is not in the document]
Comment (Sunandadadi) [reply frontend/src/app/SiteUtils/SiteUtils.tsx:189]: Agreed, will push the change

File: frontend/src/app/UserUtils/actions/FetchUserFromEmail.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sunandadadi) [reply frontend/src/app/UserUtils/actions/FetchUserFromEmail.tsx:21]: Good catch. it was supposed to be `setUserEmail('')`. Yea, I am clearing the content if the input box is empty for now. I can change it later as required.

File: frontend/src/app/UserUtils/actions/FetchUserFromEmail.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  async function fetchUser() {
+    if (userEmail != '') {
+       HttpService.axiosClient.get(`/quayuseremail/${userEmail}`)
+      .then(function (response) {
+        setResponse(response.data);
+      })
+      .catch(function (error) {
+        setMessage(error.response.data.message);
```
[the remark this answers is not in the document]
Comment (Sunandadadi) [reply frontend/src/app/UserUtils/actions/FetchUserFromEmail.tsx:32]: Added ternary operator here.

**Pull request merged**